apps/backend/src/modules/tasks/task.controller.ts

```diff
@@ -102,6 +107,85 @@ class TaskController extends BaseController {
 				body: taskUpdateValidationSchema,
 			},
 		});
+
+		this.addRoute({
+			handler: (options) =>
+				this.addNote(
+					options as APIHandlerOptions<{ body: TaskNoteRequestDto }>,
+				),
+			method: "POST",
+			path: TasksApiPath.NOTES,
+			validation: {
+				body: taskNoteValidationSchema,
+			},
+		});
+
+		this.addRoute({
+			handler: (options) =>
+				this.getNotesByTaskId(
+					options as APIHandlerOptions<{
+						params: TaskNoteParametersDto;
+					}>,
+				),
+			method: "GET",
```

Comment:
Add a preHandler to check if the user gets notes for their own tasks

Reply:
There's already a preHandler hook to check if the user has access to task when updating a task. Is it okay if I reuse that?

`const checkAccessToTask =
	(taskService: TaskService): APIPreHandler =>
	async (request) => {
		const { params, user } = request;

		const taskId = (params as TaskUpdateParametersDto).id;

		const task = await taskService.find(taskId);

		if ((user as UserDto).id !== task?.userId) {
			throw new TaskError({
				message: ErrorMessage.FORBIDDEN,
				status: HTTPCode.FORBIDDEN,
			});
		}
	};
`